devops/steps/linux/anybuild.yml

```diff
@@ -34,22 +34,27 @@ steps:
 
     readonly abClientExe="$HOME/.local/share/Microsoft/AnyBuild/AnyBuild.sh"
 
-    env -i                                                                \
-      SECRET=$(LLVMPrincipalPassword)                                     \
-      HOME="$HOME"                                                        \
-      PATH="$PATH"                                                        \
-    "$abClientExe"                                                        \
-      --RemoteExecServiceUri $(AnyBuildEnvironmentUri)                    \
-      --NoCheckForUpdates                                                 \
-      --WaitForAgentForever                                               \
-      --DoNotUseMachineUtilizationForScheduling                           \
-      --ClientApplicationId $(LLVMPrincipalAppId)                         \
-      --ClientSecretEnvironmentVariable SECRET                            \
-      --ExperimentName "${{ parameters.BuildName }}"                      \
-      --WhyCacheMiss                                                      \
-      --WhyCacheMissOptions CacheDataStoreKey=${{ parameters.BuildName }} \
-      ${{ parameters.ExtraAnyBuildArgs }}                                 \
-      --                                                                  \
+    # why cache miss key must not contain commas
+    readonly cacheDataStoreKey="${{ replace(parameters.BuildName, ',', '_' }}"
```

Comment:
The clang build passes a build name which contains commas, and commas are not allowed in a cache data store key.